File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/Enums/AllianceColor.java
package com.alphadevelopmentsolutions.frcscout.Enums;

/**
 * Colors for available alliances during the matches
 */
public enum  AllianceColor
{
    RED,
    BLUE,
    NONE;

    /**
     * Converts a string value to enum
     * @param allianceColor string enum name
     * @return enum converted from string
     */
    public static AllianceColor getColorFromString(String allianceColor)
    {
        if(allianceColor.toLowerCase().equals(RED.name().toLowerCase()))
            return RED;
        else if (allianceColor.toLowerCase().equals(BLUE.name().toLowerCase()))
            return BLUE;
        else
            return NONE;
    }
}

File: app/src/main/java/com/alphadevelopmentsolutions/frcscout/enums/AllianceColor.kt
package com.alphadevelopmentsolutions.frcscout.enums

/**
 * Colors for available alliances during the matches
 */
enum class AllianceColor
{
    RED,
    BLUE,
    NONE;


    companion object
    {

        /**
         * Converts a string value to enum
         * @param allianceColor string enum name
         * @return enum converted from string
         */
        fun getColorFromString(allianceColor: String): AllianceColor
        {
            return when
            {
                allianceColor.toLowerCase() == RED.name.toLowerCase() -> RED
                allianceColor.toLowerCase() == BLUE.name.toLowerCase() -> BLUE
                else -> NONE
            }
        }
    }
}
